public/models/OnOffTrap.js

```diff
@@ -0,0 +1,48 @@
+if (typeof exports !== 'undefined' && typeof module !== 'undefined' && module.exports) {
+    // used on the server
+    // eslint-disable-next-line global-require
+    Trap = require('./Trap').Trap;
+  }
+
+class OnOffTrap extends Trap {
+    constructor(x, y, width, height, img, value, activeTime, isActive) {
+        super(x, y, width, height, img, value);
+        this.value = value;
+        this.values = [0.001, 0.002, 0.05, 0.005];
+        this.direction = isActive;
+        this.speed = activeTime;
+
+        if (! (typeof exports !== 'undefined' && typeof module !== 'undefined' && module.exports)) {
+            const trapValue = this.value;
+            setInterval(() => {
+                if (this.direction) {
+                    this.value = 0;
+                    this.direction = !this.direction
+                } else {
+                    this.value = trapValue;
+                    this.direction = !this.direction
+                }
+            }, this.speed);
+        }
+    }
+
+    onCollision () {
+        console.log("Ow no, on off trap!");
```

Comment:
You should add removal of health here if it's active.

Reply:
nvm we got it at some other place :D But should probably think about refactoring it in the future